src/service/PromptService.ts

```diff
@@ -82,10 +106,25 @@ export async function imageUrlToBase64(url: string) {
     const buffer = Buffer.from(response.data, 'binary');
     return buffer.toString('base64');
   } catch (error) {
+    console.log(url)
+    console.log(error)
     throw new Error('Failed to convert image to base64');
   }
 }
 
+export async function removePromptModel(id: string) {
+  return await PromptModel.findByIdAndDelete(id)
+}
+
+export async function removeLatestImages(n: number) {
+  const lastImages = await PromptModel.find({})
+    .sort({ createdAt: -1 })
+    .limit(n)
+
+  const imageIds = lastImages.map(p => p._id)
+  return imageIds.forEach(async id => await PromptModel.findByIdAndDelete(id))
+}
```

Comment:
```suggestion
export async function removeLatestImages(n: number) {
  const lastImages = await PromptModel.find({})
    .sort({ createdAt: -1 })
    .limit(n);

  const imageIds = lastImages.map(p => p._id);

  if (imageIds.length > 0) {
    await PromptModel.deleteMany({ _id: { $in: imageIds } });
  }
}

```

Not sure if it works, however something like this could work as well.